File: 2707-merge-two-2d-arrays-by-summing-values/2707-merge-two-2d-arrays-by-summing-values.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        res = defaultdict(int)

        for i in range(max(len(nums1), len(nums2))):
            if i < len(nums1):
                id1, num1 = nums1[i]
                res[id1] += num1
            if i < len(nums2):
                id2, num2 =  nums2[i]
                res[id2] += num2
        return [[i, num] for i, num in sorted(res.items())]

File: 2707-merge-two-2d-arrays-by-summing-values/2707-merge-two-2d-arrays-by-summing-values.kt
class Solution {
    fun mergeArrays(nums1: Array<IntArray>, nums2: Array<IntArray>): Array<IntArray> {
        //Two pointers
        var ptr1 = 0
        var ptr2 = 0
        var res = mutableListOf<IntArray>()
        while (ptr1 < nums1.size && ptr2 < nums2.size ) {
            if (nums1[ptr1][0] == nums2[ptr2][0]) {
                var curr = nums1[ptr1][1] + nums2[ptr2][1]
                res.add(intArrayOf(nums1[ptr1][0], curr))
                ptr1++
                ptr2++
                continue
            } 
            if (nums1[ptr1][0] < nums2[ptr2][0]) {
                res.add(nums1[ptr1])
                ptr1++
            } else {
                res.add(nums2[ptr2])
                ptr2++
            }
        }

        while (ptr1 < nums1.size) {
            res.add(nums1[ptr1])
            ptr1++
        }

        while (ptr2 < nums2.size) {
            res.add(nums2[ptr2])
            ptr2++
        }
        return res.toTypedArray()
    }
}